Test Failure: com.ibm.ws.microprofile.health20.fat.ApplicationStateHealthCheckTest.testFailsToStartApplicationHealthCheckTest_mpHealth-3.0

RTC 278394

```
testFailsToStartApplicationHealthCheckTest_mpHealth-3.0:java.lang.Throwable: 2020-10-01-03:23:35:400 There were 2 errors:
  java.lang.Exception(Errors/warnings were found in server FailedApplicationStateHealthCheck logs:
<br>[10/1/20 3:18:54:737 PDT] 00000019 LogService-173-io.openliberty.io.smallrye.config             E CWWKE0702E: Could not resolve module: io.openliberty.io.smallrye.config [173]
<br>[10/1/20 3:23:00:431 PDT] 00000028 com.ibm.ws.webcontainer.util.ApplicationErrorUtils           E SRVE0777E: Exception thrown by application class 'org.eclipse.microprofile.config.spi.ConfigProviderResolver.loadSpi:134'
<br>[10/1/20 3:23:00:533 PDT] 00000028 com.ibm.ws.webcontainer.webapp                               E SRVE0315E: An exception occurred: java.lang.Throwable: java.lang.IllegalStateException: No ConfigProviderResolver implementation found!)
at componenttest.custom.junit.runner.FATRunner.newThrowableWithTimeStamp(FATRunner.java:293)
at componenttest.custom.junit.runner.FATRunner.access$700(FATRunner.java:66)
at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:265)
at componenttest.rules.repeater.RepeatTests$CompositeRepeatTestActionStatement.evaluate(RepeatTests.java:115)
at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:318)
at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:171)
  java.lang.Exception(Errors/warnings were found in server FailedApplicationStateHealthCheck logs:
<br>[10/1/20 3:18:54:737 PDT] 00000019 LogService-173-io.openliberty.io.smallrye.config             E CWWKE0702E: Could not resolve module: io.openliberty.io.smallrye.config [173]
<br>[10/1/20 3:23:00:431 PDT] 00000028 com.ibm.ws.webcontainer.util.ApplicationErrorUtils           E SRVE0777E: Exception thrown by application class 'org.eclipse.microprofile.config.spi.ConfigProviderResolver.loadSpi:134'
<br>[10/1/20 3:23:00:533 PDT] 00000028 com.ibm.ws.webcontainer.webapp                               E SRVE0315E: An exception occurred: java.lang.Throwable: java.la
```

messages.log

```
********************************************************************************
product = Open Liberty 20.0.0.11 (wlp-1.0.46.202009302120)
wlp.install.dir = /home/jazz_build/Build/jbe/build/dev/image/output/wlp/
java.home = /home/jazz_build/Build/jbe/jvms/OpenJDK11U-jdk_x64_linux_openj9_11.0.7_10_openj9-0.20.0
java.version = 11.0.7
java.runtime = OpenJDK Runtime Environment (11.0.7+10)
os = Linux (4.15.0-112-generic; amd64) (en_US)
process = 7100@10.51.8.189
********************************************************************************
[10/1/20, 6:39:58:485 PDT] 00000001 com.ibm.ws.logging.internal.TraceSpecification               I TRAS0018I: The trace state has been changed. The new trace state is *=info:com.ibm.ws.runtime.update.*=all:logservice=detail.
[10/1/20, 6:39:58:510 PDT] 00000001 com.ibm.ws.kernel.launch.internal.FrameworkManager           A CWWKE0001I: The server FailedApplicationStateHealthCheck has been launched.
[10/1/20, 6:39:59:107 PDT] 0000001d LogService-12-com.ibm.ws.org.apache.felix.scr                I CWWKE0703I: bundle com.ibm.ws.org.apache.felix.scr:1.0.46.202009302120 (12) Starting with globalExtender setting: true   
[10/1/20, 6:39:59:111 PDT] 0000001d LogService-12-com.ibm.ws.org.apache.felix.scr                I CWWKE0703I: bundle com.ibm.ws.org.apache.felix.scr:1.0.46.202009302120 (12)  Version = 1.0.46.202009302120   
[10/1/20, 6:39:59:156 PDT] 0000001e com.ibm.ws.config.xml.internal.XMLConfigParser               A CWWKG0028A: Processing included configuration resource: /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/fatTestPorts.xml
[10/1/20, 6:39:59:158 PDT] 0000001e com.ibm.ws.config.xml.internal.XMLConfigParser               A CWWKG0028A: Processing included configuration resource: /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/fatTestCommon.xml
[10/1/20, 6:39:59:406 PDT] 00000023 com.ibm.ws.logging.internal.TraceSpecification               I TRAS0018I: The trace state has been changed. The new trace state is *=info:HEALTH=all:com.ibm.websphere.org.eclipse.microprofile.health.2.0.*=all:logservice=all.
[10/1/20, 6:40:00:159 PDT] 00000001 com.ibm.ws.kernel.launch.internal.FrameworkManager           I CWWKE0002I: The kernel started after 1.861 seconds
[10/1/20, 6:40:00:322 PDT] 00000027 com.ibm.ws.kernel.feature.internal.FeatureManager            I CWWKF0007I: Feature update started.
[10/1/20, 6:40:09:487 PDT] 0000001c com.ibm.ws.security.ready.internal.SecurityReadyServiceImpl  I CWWKS0007I: The security service is starting...
[10/1/20, 6:40:10:071 PDT] 0000001e com.ibm.ws.app.manager.internal.monitor.DropinMonitor        A CWWKZ0058I: Monitoring dropins for applications.
[10/1/20, 6:40:10:694 PDT] 00000018 LogService-173-io.openliberty.io.smallrye.config             E CWWKE0702E: Could not resolve module: io.openliberty.io.smallrye.config [173]
  Unresolved requirement: Require-Capability: osgi.cdi.extension; filter:="(osgi.cdi.extension=org.eclipse.microprofile.config)"; resolution:="optional"
  Bundle was not resolved because of a uses constraint violation.
  org.apache.felix.resolver.reason.ReasonException: Uses constraint violation. Unable to resolve resource io.openliberty.io.smallrye.config [osgi.identity; osgi.identity="io.openliberty.io.smallrye.config"; type="osgi.bundle"; version:Version="1.0.46.202009302120"; singleton:="true"] because it is exposed to package 'org.eclipse.microprofile.config' from resources io.openliberty.org.eclipse.microprofile.config.2.0 [osgi.identity; osgi.identity="io.openliberty.org.eclipse.microprofile.config.2.0"; type="osgi.bundle"; version:Version="1.0.46.202009302120"; singleton:="true"] and com.ibm.websphere.org.eclipse.microprofile.config.1.4 [osgi.identity; osgi.identity="com.ibm.websphere.org.eclipse.microprofile.config.1.4"; type="osgi.bundle"; version:Version="1.0.46.202009302120"; singleton:="true"] via two dependency chains.

Chain 1:
  io.openliberty.io.smallrye.config [osgi.identity; osgi.identity="io.openliberty.io.smallrye.config"; type="osgi.bundle"; version:Version="1.0.46.202009302120"; singleton:="true"]
    import: (&(osgi.wiring.package=org.eclipse.microprofile.config)(&(version>=2.0.0)(!(version>=2.1.0))))
     |
    export: osgi.wiring.package: org.eclipse.microprofile.config
  io.openliberty.org.eclipse.microprofile.config.2.0 [osgi.identity; osgi.identity="io.openliberty.org.eclipse.microprofile.config.2.0"; type="osgi.bundle"; version:Version="1.0.46.202009302120"; singleton:="true"]

Chain 2:
  io.openliberty.io.smallrye.config [osgi.identity; osgi.identity="io.openliberty.io.smallrye.config"; type="osgi.bundle"; version:Version="1.0.46.202009302120"; singleton:="true"]
    import: (&(osgi.wiring.package=io.openliberty.microprofile.config.internal.serverxml)(&(version>=1.0.0)(!(version>=2.0.0))))
     |
    export: osgi.wiring.package=io.openliberty.microprofile.config.internal.serverxml; uses:=io.openliberty.microprofile.config.internal.common
  io.openliberty.microprofile.config.internal.serverxml [osgi.identity; osgi.identity="io.openliberty.microprofile.config.internal.serverxml"; type="osgi.bundle"; version:Version="1.0.46.202009302120"]
    import: (&(osgi.wiring.package=io.openliberty.microprofile.config.internal.common)(&(version>=1.0.0)(!(version>=2.0.0))))
     |
    export: osgi.wiring.package=io.openliberty.microprofile.config.internal.common; uses:=org.eclipse.microprofile.config.spi
  io.openliberty.microprofile.config.internal.common [osgi.identity; osgi.identity="io.openliberty.microprofile.config.internal.common"; type="osgi.bundle"; version:Version="1.0.46.202009302120"]
    import: (&(osgi.wiring.package=org.eclipse.microprofile.config.spi)(&(version>=1.0.0)(!(version>=3.0.0))))
     |
    export: osgi.wiring.package: org.eclipse.microprofile.config.spi; uses:=org.eclipse.microprofile.config
    export: osgi.wiring.package=org.eclipse.microprofile.config
  com.ibm.websphere.org.eclipse.microprofile.config.1.4 [osgi.identity; osgi.identity="com.ibm.websphere.org.eclipse.microprofile.config.1.4"; type="osgi.bundle"; version:Version="1.0.46.202009302120"; singleton:="true"]
[10/1/20, 6:40:10:931 PDT] 0000001d com.ibm.ws.cache.ServerCache                                 I DYNA1001I: WebSphere Dynamic Cache instance named baseCache initialized successfully.
[10/1/20, 6:40:10:933 PDT] 0000001d com.ibm.ws.cache.ServerCache                                 I DYNA1071I: The cache provider default is being used.
[10/1/20, 6:40:10:934 PDT] 0000001d com.ibm.ws.cache.CacheServiceImpl                            I DYNA1056I: Dynamic Cache (object cache) initialized successfully.
[10/1/20, 6:40:11:064 PDT] 00000023 com.ibm.ws.ssl.config.WSKeyStore                             I CWPKI0819I: The default keystore is not created because a password is not configured on the <keyStore id="defaultKeyStore"/> element, and the 'keystore_password' environment variable is not set. 
[10/1/20, 6:40:11:104 PDT] 00000029 com.ibm.ws.security.token.ltpa.internal.LTPAKeyInfoManager   I CWWKS4103I: Creating the LTPA keys. This may take a few seconds.
[10/1/20, 6:40:11:166 PDT] 0000001c ibm.ws.security.authentication.internal.jaas.JAASServiceImpl I CWWKS1123I: The collective authentication plugin with class name NullCollectiveAuthenticationPlugin has been activated. 
[10/1/20, 6:40:11:645 PDT] 0000001c com.ibm.ws.timedexit.internal.TimedExitComponent             A TE9900A: Timed Exit Enabled
[10/1/20, 6:40:12:219 PDT] 0000002a com.ibm.ws.app.manager.AppMessageHelper                      I CWWKZ0018I: Starting application FailsToStartHealthCheckApp.
[10/1/20, 6:40:12:220 PDT] 0000002a bm.ws.app.manager.war.internal.WARDeployedAppInfoFactoryImpl I CWWKZ0136I: The FailsToStartHealthCheckApp application is using the archive file at the /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/FailedApplicationStateHealthCheck/apps/FailsToStartHealthCheckApp.war location.
[10/1/20, 6:40:12:572 PDT] 00000029 com.ibm.ws.security.token.ltpa.internal.LTPAKeyInfoManager   A CWWKS4104A: LTPA keys created in 1.468 seconds. LTPA key file: /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/FailedApplicationStateHealthCheck/resources/security/ltpa.keys
[10/1/20, 6:40:12:576 PDT] 00000029 com.ibm.ws.security.token.ltpa.internal.LTPAKeyCreateTask    I CWWKS4105I: LTPA configuration is ready after 1.473 seconds.
[10/1/20, 6:40:12:704 PDT] 00000028 com.ibm.ws.webcontainer.osgi.webapp.WebGroup                 I SRVE0169I: Loading Web Module: health.
[10/1/20, 6:40:12:707 PDT] 00000028 com.ibm.ws.webcontainer                                      I SRVE0250I: Web Module health has been bound to default_host.
[10/1/20, 6:40:12:750 PDT] 00000028 com.ibm.ws.http.internal.VirtualHostImpl                     A CWWKT0016I: Web application available (default_host): http://ebcprh0434fquogu3-n.fyre.ibm.com:8010/health/
[10/1/20, 6:40:12:799 PDT] 00000028 com.ibm.ws.session.WASSessionCore                            I SESN8501I: The session manager did not find a persistent storage location; HttpSession objects will be stored in the local application server's memory.
[10/1/20, 6:40:12:821 PDT] 0000002a org.jboss.weld.Version                                       I WELD-000900: 3.1.4 (Final)
[10/1/20, 6:40:12:843 PDT] 00000029 com.ibm.ws.session.WASSessionCore                            I SESN0176I: A new session context will be created for application key default_host/health
[10/1/20, 6:40:12:889 PDT] 00000029 com.ibm.ws.util                                              I SESN0172I: The session manager is using the Java default SecureRandom implementation for session ID generation.
[10/1/20, 6:40:12:957 PDT] 00000029 com.ibm.ws.cache.CacheServiceImpl                            I DYNA1056I: Dynamic Cache (object cache) initialized successfully.
[10/1/20, 6:40:13:107 PDT] 00000029 com.ibm.ws.webcontainer.servlet                              I SRVE0242I: [io.openliberty.microprofile.health.3.0.internal] [/health] [HealthCheckReadinessServlet]: Initialization successful.
[10/1/20, 6:40:13:147 PDT] 00000029 com.ibm.ws.webcontainer.servlet                              I SRVE0242I: [io.openliberty.microprofile.health.3.0.internal] [/health] [HealthCheckServlet]: Initialization successful.
[10/1/20, 6:40:13:191 PDT] 00000029 com.ibm.ws.webcontainer.servlet                              I SRVE0242I: [io.openliberty.microprofile.health.3.0.internal] [/health] [HealthCheckLivenessServlet]: Initialization successful.
[10/1/20, 6:40:13:610 PDT] 00000028 com.ibm.ws.webcontainer.osgi.mbeans.PluginGenerator          I SRVE9103I: A configuration file for a web server plugin was automatically generated for this server at /home/jazz_build/Build/jbe/build/dev/image/output/wlp/usr/servers/FailedApplicationStateHealthCheck/logs/state/plugin-cfg.xml.
[10/1/20, 6:40:14:660 PDT] 0000002a lrye.reactive.messaging.extension.ReactiveMessagingExtension I Analyzing mediator bean: Managed Bean [class com.ibm.ws.microprofile.health20.fails.to.start.health.check.app.BasicMessagingBean] with qualifiers [@Any @Default]
[10/1/20, 6:40:14:661 PDT] 0000002a io.smallrye.reactive.messaging.extension.MediatorManager     I Scanning Type: class com.ibm.ws.microprofile.health20.fails.to.start.health.check.app.BasicMessagingBean
[10/1/20, 6:40:14:669 PDT] 0000002a io.smallrye.reactive.messaging.extension.MediatorManager     I Deployment done... start processing
[10/1/20, 6:40:14:686 PDT] 0000002a io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory I Found incoming connectors: []
[10/1/20, 6:40:14:687 PDT] 0000002a io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory I Found outgoing connectors: []
[10/1/20, 6:40:14:687 PDT] 0000002a io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory I Stream manager initializing...
[10/1/20, 6:40:14:852 PDT] 0000002a org.apache.kafka.clients.consumer.ConsumerConfig             I ConsumerConfig values: 
	allow.auto.create.topics = true
	auto.commit.interval.ms = 5000
	auto.offset.reset = latest
	bootstrap.servers = [this.is.not.going-to-resolve:9092]
	check.crcs = true
	client.dns.lookup = default
	client.id = 
	client.rack = 
	connections.max.idle.ms = 540000
	default.api.timeout.ms = 60000
	enable.auto.commit = false
	exclude.internal.topics = true
	fetch.max.bytes = 52428800
	fetch.max.wait.ms = 500
	fetch.min.bytes = 1
	group.id = failstostart
	group.instance.id = null
	heartbeat.interval.ms = 3000
	interceptor.classes = []
	internal.leave.group.on.close = true
	isolation.level = read_uncommitted
	key.deserializer = class org.apache.kafka.common.serialization.StringDeserializer
	max.partition.fetch.bytes = 1048576
	max.poll.interval.ms = 300000
	max.poll.records = 500
	metadata.max.age.ms = 300000
	metric.reporters = []
	metrics.num.samples = 2
	metrics.recording.level = INFO
	metrics.sample.window.ms = 30000
	partition.assignment.strategy = [class org.apache.kafka.clients.consumer.RangeAssignor]
	receive.buffer.bytes = 65536
	reconnect.backoff.max.ms = 1000
	reconnect.backoff.ms = 50
	request.timeout.ms = 30000
	retry.backoff.ms = 100
	sasl.client.callback.handler.class = null
	sasl.jaas.config = null
	sasl.kerberos.kinit.cmd = /usr/bin/kinit
	sasl.kerberos.min.time.before.relogin = 60000
	sasl.kerberos.service.name = null
	sasl.kerberos.ticket.renew.jitter = 0.05
	sasl.kerberos.ticket.renew.window.factor = 0.8
	sasl.login.callback.handler.class = null
	sasl.login.class = null
	sasl.login.refresh.buffer.seconds = 300
	sasl.login.refresh.min.period.seconds = 60
	sasl.login.refresh.window.factor = 0.8
	sasl.login.refresh.window.jitter = 0.05
	sasl.mechanism = GSSAPI
	security.protocol = PLAINTEXT
	send.buffer.bytes = 131072
	session.timeout.ms = 10000
	ssl.cipher.suites = null
	ssl.enabled.protocols = [TLSv1.2, TLSv1.1, TLSv1]
	ssl.endpoint.identification.algorithm = https
	ssl.key.password = null
	ssl.keymanager.algorithm = SunX509
	ssl.keystore.location = null
	ssl.keystore.password = null
	ssl.keystore.type = JKS
	ssl.protocol = TLS
	ssl.provider = null
	ssl.secure.random.implementation = null
	ssl.trustmanager.algorithm = PKIX
	ssl.truststore.location = null
	ssl.truststore.password = null
	ssl.truststore.type = JKS
	value.deserializer = class org.apache.kafka.common.serialization.StringDeserializer

[10/1/20, 6:40:14:891 PDT] 0000002a org.apache.kafka.clients.ClientUtils                         W Couldn't resolve server this.is.not.going-to-resolve:9092 from bootstrap.servers as DNS resolution failed for this.is.not.going-to-resolve
[10/1/20, 6:40:14:925 PDT] 0000002a com.ibm.ws.logging.internal.impl.IncidentImpl                I FFDC1015I: An FFDC Incident has been created: "com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.KafkaAdapterException: org.apache.kafka.common.KafkaException: Failed to construct kafka consumer: org.apache.kafka.common.config.ConfigException: No resolvable bootstrap urls given in bootstrap.servers com.ibm.ws.microprofile.reactive.messaging.kafka.KafkaIncomingConnector 137" at ffdc_20.10.01_06.40.14.0.log
[10/1/20, 6:40:14:928 PDT] 0000002a io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory E Unable to create the publisher or subscriber during initialization
com.ibm.ws.microprofile.reactive.messaging.kafka.KafkaConnectorException: CWMRX1007E: A Kafka incoming connector for the test-out channel cannot be created. The error is org.apache.kafka.common.KafkaException: Failed to construct kafka consumer: org.apache.kafka.common.config.ConfigException: No resolvable bootstrap urls given in bootstrap.servers
	at com.ibm.ws.microprofile.reactive.messaging.kafka.KafkaIncomingConnector.getPublisherBuilder(KafkaIncomingConnector.java:138)
	at com.ibm.ws.microprofile.reactive.messaging.kafka.KafkaIncomingConnector$Proxy$_$$_WeldClientProxy.getPublisherBuilder(Unknown Source)
	at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory.createPublisherBuilder(ConfiguredChannelFactory.java:145)
	at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory.lambda$register$4(ConfiguredChannelFactory.java:123)
	at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory$$Lambda$285/0000000000000000.accept(Unknown Source)
	at java.base/java.util.HashMap.forEach(HashMap.java:1336)
	at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory.register(ConfiguredChannelFactory.java:123)
	at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory.initialize(ConfiguredChannelFactory.java:118)
	at io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory$Proxy$_$$_WeldClientProxy.initialize(Unknown Source)
	at io.smallrye.reactive.messaging.extension.MediatorManager$$Lambda$280/0000000000000000.accept(Unknown Source)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
	at io.smallrye.reactive.messaging.extension.MediatorManager.initializeAndRun(MediatorManager.java:132)
	at io.smallrye.reactive.messaging.extension.MediatorManager$Proxy$_$$_WeldClientProxy.initializeAndRun(Unknown Source)
	at io.smallrye.reactive.messaging.extension.ReactiveMessagingExtension.afterDeploymentValidation(ReactiveMessagingExtension.java:86)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:95)
	at org.jboss.weld.injection.MethodInvocationStrategy$SpecialParamPlusBeanManagerStrategy.invoke(MethodInvocationStrategy.java:187)
	at org.jboss.weld.event.ObserverMethodImpl.sendEvent(ObserverMethodImpl.java:330)
	at org.jboss.weld.event.ExtensionObserverMethodImpl.sendEvent(ExtensionObserverMethodImpl.java:123)
	at org.jboss.weld.event.ObserverMethodImpl.sendEvent(ObserverMethodImpl.java:308)
	at org.jboss.weld.event.ObserverMethodImpl.notify(ObserverMethodImpl.java:286)
	at javax.enterprise.inject.spi.ObserverMethod.notify(ObserverMethod.java:124)
	at org.jboss.weld.util.Observers.notify(Observers.java:166)
	at org.jboss.weld.event.ObserverNotifier.notifySyncObservers(ObserverNotifier.java:285)
	at org.jboss.weld.event.ObserverNotifier.notify(ObserverNotifier.java:273)
	at org.jboss.weld.event.ObserverNotifier.fireEvent(ObserverNotifier.java:177)
	at org.jboss.weld.event.ObserverNotifier.fireEvent(ObserverNotifier.java:171)
	at org.jboss.weld.bootstrap.events.AbstractContainerEvent.fire(AbstractContainerEvent.java:53)
	at org.jboss.weld.bootstrap.events.AbstractDeploymentContainerEvent.fire(AbstractDeploymentContainerEvent.java:35)
	at org.jboss.weld.bootstrap.events.AfterDeploymentValidationImpl.fire(AfterDeploymentValidationImpl.java:28)
	at org.jboss.weld.bootstrap.WeldStartup.validateBeans(WeldStartup.java:505)
	at org.jboss.weld.bootstrap.WeldBootstrap.validateBeans(WeldBootstrap.java:93)
	at com.ibm.ws.cdi.impl.CDIContainerImpl.startInitialization(CDIContainerImpl.java:161)
	at com.ibm.ws.cdi.liberty.CDIRuntimeImpl.applicationStarting(CDIRuntimeImpl.java:479)
	at com.ibm.ws.container.service.state.internal.ApplicationStateManager.fireStarting(ApplicationStateManager.java:51)
	at com.ibm.ws.container.service.state.internal.StateChangeServiceImpl.fireApplicationStarting(StateChangeServiceImpl.java:50)
	at com.ibm.ws.app.manager.module.internal.SimpleDeployedAppInfoBase.preDeployApp(SimpleDeployedAppInfoBase.java:547)
	at com.ibm.ws.app.manager.module.internal.SimpleDeployedAppInfoBase.installApp(SimpleDeployedAppInfoBase.java:508)
	at com.ibm.ws.app.manager.module.internal.DeployedAppInfoBase.deployApp(DeployedAppInfoBase.java:349)
	at com.ibm.ws.app.manager.war.internal.WARApplicationHandlerImpl.install(WARApplicationHandlerImpl.java:65)
	at com.ibm.ws.app.manager.internal.statemachine.StartAction.execute(StartAction.java:144)
	at com.ibm.ws.app.manager.internal.statemachine.ApplicationStateMachineImpl.enterState(ApplicationStateMachineImpl.java:1317)
	at com.ibm.ws.app.manager.internal.statemachine.ApplicationStateMachineImpl.run(ApplicationStateMachineImpl.java:897)
	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:239)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.KafkaAdapterException: org.apache.kafka.common.KafkaException: Failed to construct kafka consumer: org.apache.kafka.common.config.ConfigException: No resolvable bootstrap urls given in bootstrap.servers
	at com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.KafkaAdapterFactory.getInstance(KafkaAdapterFactory.java:182)
	at com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.KafkaAdapterFactory.getInstance(KafkaAdapterFactory.java:168)
	at com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.KafkaAdapterFactory.newKafkaConsumer(KafkaAdapterFactory.java:98)
	at com.ibm.ws.microprofile.reactive.messaging.kafka.AdapterFactoryImpl$Proxy$_$$_WeldClientProxy.newKafkaConsumer(Unknown Source)
	at com.ibm.ws.microprofile.reactive.messaging.kafka.KafkaIncomingConnector.getKafkaConsumerWithRetry(KafkaIncomingConnector.java:145)
	at com.ibm.ws.microprofile.reactive.messaging.kafka.KafkaIncomingConnector.getPublisherBuilder(KafkaIncomingConnector.java:120)
	... 51 more
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.KafkaAdapterFactory.getInstance(KafkaAdapterFactory.java:178)
	... 56 more
Caused by: org.apache.kafka.common.KafkaException: Failed to construct kafka consumer
	at org.apache.kafka.clients.consumer.KafkaConsumer.<init>(KafkaConsumer.java:827)
	at org.apache.kafka.clients.consumer.KafkaConsumer.<init>(KafkaConsumer.java:629)
	at org.apache.kafka.clients.consumer.KafkaConsumer.<init>(KafkaConsumer.java:610)
	at com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.impl.KafkaConsumerImpl.<init>(KafkaConsumerImpl.java:37)
	... 61 more
Caused by: org.apache.kafka.common.config.ConfigException: No resolvable bootstrap urls given in bootstrap.servers
	at org.apache.kafka.clients.ClientUtils.parseAndValidateAddresses(ClientUtils.java:88)
	at org.apache.kafka.clients.ClientUtils.parseAndValidateAddresses(ClientUtils.java:47)
	at org.apache.kafka.clients.consumer.KafkaConsumer.<init>(KafkaConsumer.java:735)
	... 64 more

[10/1/20, 6:40:14:954 PDT] 0000002a com.ibm.ws.logging.internal.impl.IncidentImpl                I FFDC1015I: An FFDC Incident has been created: "org.jboss.weld.exceptions.DeploymentException: org.apache.kafka.common.KafkaException: Failed to construct kafka consumer: org.apache.kafka.common.config.ConfigException: No resolvable bootstrap urls given in bootstrap.servers com.ibm.ws.container.service.state.internal.ApplicationStateManager 54" at ffdc_20.10.01_06.40.14.1.log
[10/1/20, 6:40:14:995 PDT] 0000002a com.ibm.ws.logging.internal.impl.IncidentImpl                I FFDC1015I: An FFDC Incident has been created: "com.ibm.ws.container.service.state.StateChangeException: org.jboss.weld.exceptions.DeploymentException: org.apache.kafka.common.KafkaException: Failed to construct kafka consumer: org.apache.kafka.common.config.ConfigException: No resolvable bootstrap urls given in bootstrap.servers com.ibm.ws.app.manager.module.internal.SimpleDeployedAppInfoBase 548" at ffdc_20.10.01_06.40.14.2.log
[10/1/20, 6:40:15:000 PDT] 0000002a io.smallrye.reactive.messaging.extension.MediatorManager     I Cancel subscriptions
[10/1/20, 6:40:15:006 PDT] 0000002a com.ibm.ws.app.manager.AppMessageHelper                      E CWWKZ0002E: An exception occurred while starting the application FailsToStartHealthCheckApp. The exception message was: com.ibm.ws.container.service.state.StateChangeException: org.jboss.weld.exceptions.DeploymentException: org.apache.kafka.common.KafkaException: Failed to construct kafka consumer: org.apache.kafka.common.config.ConfigException: No resolvable bootstrap urls given in bootstrap.servers
[10/1/20, 6:40:15:112 PDT] 00000027 com.ibm.ws.tcpchannel.internal.TCPPort                       I CWWKO0219I: TCP Channel defaultHttpEndpoint has been started and is now listening for requests on host *  (IPv4) port 8010.
[10/1/20, 6:40:15:171 PDT] 00000027 com.ibm.ws.kernel.feature.internal.FeatureManager            A CWWKF0012I: The server installed the following features: [appSecurity-2.0, cdi-2.0, componenttest-1.0, concurrent-1.0, distributedMap-1.0, jndi-1.0, json-1.0, localConnector-1.0, mpConfig-1.4, mpConfig-2.0, mpHealth-3.0, mpReactiveMessaging-1.0, mpReactiveStreams-1.0, osgiConsole-1.0, servlet-4.0, ssl-1.0, timedexit-1.0].
[10/1/20, 6:40:15:172 PDT] 00000027 com.ibm.ws.kernel.feature.internal.FeatureManager            I CWWKF0008I: Feature update completed in 15.012 seconds.
[10/1/20, 6:40:15:172 PDT] 00000027 com.ibm.ws.kernel.feature.internal.FeatureManager            A CWWKF0011I: The FailedApplicationStateHealthCheck server is ready to run a smarter planet. The FailedApplicationStateHealthCheck server started in 16.877 seconds.
[10/1/20, 6:44:16:619 PDT] 0000002a com.ibm.ws.webcontainer.util.ApplicationErrorUtils           E SRVE0777E: Exception thrown by application class 'org.eclipse.microprofile.config.spi.ConfigProviderResolver.loadSpi:134'
java.lang.IllegalStateException: No ConfigProviderResolver implementation found!
	at org.eclipse.microprofile.config.spi.ConfigProviderResolver.loadSpi(ConfigProviderResolver.java:134)
	at org.eclipse.microprofile.config.spi.ConfigProviderResolver.instance(ConfigProviderResolver.java:120)
	at org.eclipse.microprofile.config.ConfigProvider.getConfig(ConfigProvider.java:85)
	at io.openliberty.microprofile.health30.internal.HealthCheck30ServiceImpl.performHealthCheck(HealthCheck30ServiceImpl.java:107)
	at io.openliberty.microprofile.health30.internal.servlet.HealthCheckLivenessServlet.service(HealthCheckLivenessServlet.java:44)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:791)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1230)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:729)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:426)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.invokeTarget(WebAppFilterChain.java:182)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:93)
	at com.ibm.ws.app.manager.wab.internal.OsgiDirectoryProtectionFilter.doFilter(OsgiDirectoryProtectionFilter.java:90)
	at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:201)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:90)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.doFilter(WebAppFilterManager.java:1001)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1139)
	at com.ibm.ws.webcontainer.webapp.WebApp.handleRequest(WebApp.java:5057)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.handleRequest(DynamicVirtualHost.java:314)
	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:1005)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:279)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1134)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:415)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:374)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:577)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:511)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:359)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:326)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:167)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:75)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:504)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:574)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:958)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1047)
	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:239)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

[10/1/20, 6:44:16:732 PDT] 0000002a com.ibm.ws.logging.internal.impl.IncidentImpl                I FFDC1015I: An FFDC Incident has been created: "java.lang.IllegalStateException: No ConfigProviderResolver implementation found! com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters 1105" at ffdc_20.10.01_06.44.16.0.log
[10/1/20, 6:44:16:736 PDT] 0000002a com.ibm.ws.webcontainer.webapp                               E SRVE0315E: An exception occurred: java.lang.Throwable: java.lang.IllegalStateException: No ConfigProviderResolver implementation found!
	at com.ibm.ws.webcontainer.webapp.WebApp.handleRequest(WebApp.java:5115)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.handleRequest(DynamicVirtualHost.java:314)
	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:1005)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:279)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1134)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:415)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:374)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:577)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:511)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:359)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:326)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:167)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:75)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:504)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:574)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:958)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1047)
	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:239)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalStateException: No ConfigProviderResolver implementation found!
	at org.eclipse.microprofile.config.spi.ConfigProviderResolver.loadSpi(ConfigProviderResolver.java:134)
	at org.eclipse.microprofile.config.spi.ConfigProviderResolver.instance(ConfigProviderResolver.java:120)
	at org.eclipse.microprofile.config.ConfigProvider.getConfig(ConfigProvider.java:85)
	at io.openliberty.microprofile.health30.internal.HealthCheck30ServiceImpl.performHealthCheck(HealthCheck30ServiceImpl.java:107)
	at io.openliberty.microprofile.health30.internal.servlet.HealthCheckLivenessServlet.service(HealthCheckLivenessServlet.java:44)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:791)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1230)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:729)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:426)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.invokeTarget(WebAppFilterChain.java:182)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:93)
	at com.ibm.ws.app.manager.wab.internal.OsgiDirectoryProtectionFilter.doFilter(OsgiDirectoryProtectionFilter.java:90)
	at com.ibm.ws.webcontainer.filter.FilterInstanceWrapper.doFilter(FilterInstanceWrapper.java:201)
	at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:90)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.doFilter(WebAppFilterManager.java:1001)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1139)
	at com.ibm.ws.webcontainer.webapp.WebApp.handleRequest(WebApp.java:5057)
	... 20 more

[10/1/20, 6:44:48:034 PDT] 0000001a com.ibm.ws.kernel.launch.internal.ServerCommandListener      A CWWKE0055I: Server shutdown requested on Thursday, October 1, 2020 at 6:44 AM. The server FailedApplicationStateHealthCheck is shutting down.
[10/1/20, 6:44:48:038 PDT] 00000040 com.ibm.ws.runtime.update.internal.RuntimeUpdateManagerImpl  A CWWKE1100I: Waiting for up to 30 seconds for the server to quiesce.
[10/1/20, 6:44:48:044 PDT] 0000003b com.ibm.ws.tcpchannel.internal.TCPChannel                    I CWWKO0220I: TCP Channel defaultHttpEndpoint has stopped listening for requests on host *  (IPv4) port 8010.
[10/1/20, 6:44:48:044 PDT] 0000003b com.ibm.ws.http.internal.VirtualHostImpl                     A CWWKT0017I: Web application removed (default_host): http://ebcprh0434fquogu3-n.fyre.ibm.com:8010/health/
[10/1/20, 6:44:49:046 PDT] 00000040 com.ibm.ws.runtime.update.internal.RuntimeUpdateManagerImpl  I CWWKE1101I: Server quiesce complete.
[10/1/20, 6:44:49:102 PDT] 00000040 com.ibm.ws.webcontainer.servlet                              I SRVE0253I: [io.openliberty.microprofile.health.3.0.internal] [/health] [HealthCheckReadinessServlet]: Destroy successful.
[10/1/20, 6:44:49:103 PDT] 00000040 com.ibm.ws.webcontainer.servlet                              I SRVE0253I: [io.openliberty.microprofile.health.3.0.internal] [/health] [HealthCheckLivenessServlet]: Destroy successful.
[10/1/20, 6:44:49:103 PDT] 00000040 com.ibm.ws.webcontainer.servlet                              I SRVE0253I: [io.openliberty.microprofile.health.3.0.internal] [/health] [HealthCheckServlet]: Destroy successful.
[10/1/20, 6:44:49:302 PDT] 00000040 com.ibm.ws.security.ready.internal.SecurityReadyServiceImpl  I CWWKS0009I: The security service has stopped.
[10/1/20, 6:44:49:371 PDT] 00000040 ibm.ws.security.authentication.internal.jaas.JAASServiceImpl I CWWKS1124I: The collective authentication plugin with class name NullCollectiveAuthenticationPlugin has been deactivated. 
[10/1/20, 6:44:50:410 PDT] 00000001 com.ibm.ws.kernel.launch.internal.FrameworkManager           A CWWKE0036I: The server FailedApplicationStateHealthCheck stopped after 4 minutes, 52.115 seconds.

```

--- a/dev/com.ibm.ws.microprofile.health.2.0_fat/fat/src/com/ibm/ws/microprofile/health20/fat/ApplicationStateHealthCheckTest.java
+++ b/dev/com.ibm.ws.microprofile.health.2.0_fat/fat/src/com/ibm/ws/microprofile/health20/fat/ApplicationStateHealthCheckTest.java
@@ -42,6 +42,7 @@ import com.ibm.websphere.simplicity.log.Log;
 
 import componenttest.annotation.ExpectedFFDC;
 import componenttest.annotation.Server;
+import componenttest.annotation.SkipForRepeat;
 import componenttest.custom.junit.runner.FATRunner;
 import componenttest.custom.junit.runner.Mode;
 import componenttest.custom.junit.runner.Mode.TestMode;
@@ -94,7 +95,7 @@ public class ApplicationStateHealthCheckTest {
                                     .withID("mpHealth-3.0")
                                     .addFeature("mpHealth-3.0")
                                     .removeFeature("mpHealth-2.0")
-                                    .forServers(SERVER_NAME, FAILS_TO_START_SERVER_NAME));
+                                    .forServers(SERVER_NAME));
 
     @Server(SERVER_NAME)
     public static LibertyServer server1;
@@ -125,9 +126,13 @@ public class ApplicationStateHealthCheckTest {
      * This test will first load an application that purposely fails to start.
      * It will then load a dropin that would like to reports UP on all health checks.
      * But since the pre-loaded app failed to start, readiness/overall reports DOWN and liveness remains unaltered.
+     *
+     * Note: Not repeated for mpHealth-3.0, since mpReactiveMessaging-1.0 uses mpConfig-1.4, which conflicts with mpConfig-2.0 from
+     * mpHealth-3.0. Will allow for mpHealth-3.x repetition, once mpReactiveMessaging starts using mpConfig-2.0.
      */
     @Test
     @Mode(TestMode.FULL)
+    @SkipForRepeat("mpHealth-3.0")
     @ExpectedFFDC({ "com.ibm.ws.container.service.state.StateChangeException",
                     "com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.KafkaAdapterException",
                     "org.jboss.weld.exceptions.DeploymentException" })

